### PR TITLE
Compile-Time Resolver Validations

### PR DESCRIPTION
**Pull request by zkayser**
This PR adds an extra compile-time check on ensuring that resolver function references passed by users actually point to valid, existing functions.

Passing an undefined function to a resolver will now fail at compile time and provide users with feedback that notifies them of the error and how to fix it.

## Validations:

- A regression test is provided here https://github.com/carsdotcom/pacer/commit/b560b4c7d47b51062ba052428f11a93120700cd5. Check out this commit to see the test fail, then check out the latest from this branch and re-run the same test.

- Test this against some of your own Workflows

## Additional Notes

This is a step in the right direction, but incomplete. There are a class of problems that would lead to `FunctionClauseError`,  etc., that we cannot definitively catch at compile time. A good example of this would be if users provide a valid MFA reference for a resolver, but the definition of the function has specific pattern-matches that are incompatible with or not comprehensive enough to match the dependencies defined by a field in a `Pacer.Workflow`.

As a result, one idea we have is to allow for a `debug` mode option that can be opted into for `Pacer.Workflows` (and maybe set as a default in dev environments). In `debug` mode, we would log out any exceptions caught so that Pacer does not just silently swallow those types of errors. Behavior-wise, Pacer will still maintain the same behavior, but it will be more explicit for users when errors do happen at runtime.

### CODE REVIEW

File: lib/workflow.ex [hunk 1]
```diff
@@ -905,10 +905,27 @@ defmodule Pacer.Workflow do
...
   def __after_compile__(%{module: module} = _env, _) do
     _ = validate_dependencies(module)
+    _ = validate_resolvers(module)
 
     :ok
   end
 
+  defp validate_resolvers(module) do
+    for {field, resolver_fun} <- Module.get_attribute(module, :pacer_resolvers) do
+      Enum.into(Function.info(resolver_fun), %{})
```
Comment (stelane) [inline lib/workflow.ex:915]: can/should we pipe this even further? 
```elixir
resolver_fun
|> Function.info()
|> Enum.into(%{})
|> then(...
```

Comment (zkayser) [reply lib/workflow.ex:915]: Yes we can! Coming soon.

**Review by robert-giles — APPROVED** [review]
this is sweet! offering this info will be helpful

**Review by stelane — APPROVED** [review]
I like this a lot! thanks for putting this up!

File: test/workflow_test.exs [hunk 2]
```diff
@@ -595,6 +595,30 @@ defmodule Pacer.WorkflowTest do
         Code.eval_string(module)
       end
     end
+
+    test "alerts users when a resolver function does not exist" do
```
Comment (ethangunderson) [inline test/workflow_test.exs:599]: Maybe this distinction doesn't really matter, but do we want a test that validates when the module doesn't exist vs when the module exists but the function doesn't?

File: lib/workflow.ex (hunk 1 above)
Comment (zkayser) [reply lib/workflow.ex]: Updated to a pipeline chain here https://github.com/carsdotcom/pacer/pull/16/commits/414bef59fb5c9004ba8ebc3f0ab50b830329d8de

File: test/workflow_test.exs (hunk 2 above)
Comment (zkayser) [reply test/workflow_test.exs:599]: This turned out to be a tricker problem to determine if a module actually exists or not at compile-time. I think we could probably find a way to do it if we spend some more time, but I'm not sure if the distinction matters all that much as long as we're alerting users if they gave us an invalid function definition.

**Pull request merged**